components/timestamp/timestamp.tsx

```diff
@@ -343,6 +351,14 @@ class Timestamp extends PureComponent<Props, State> {
         }
     }
 
+    static getDerivedStateFromProps(props: Props, state: State) {
+        if (props.value !== state.prevValue) {
+            return ({now: new Date(), prevValue: props.value});
```

Comment:
Can you elaborate on the need to touch `now` here? Does this have side effect on other uses of this component?

Reply:
Most of the time `value` is older than `now` e.g. a fixed timestamp with an ongoing `now` that updates on an interval.

But when a new `value` comes down the pipe, this mechanism ensures that `now` isn't stale, preventing incorrect diffs.

I think I could and probably should move this to a separate PR.